src/TimeModeling/Types/judiWeights.jl

```diff
@@ -42,41 +42,25 @@ Construct vector cell array of weights. The `weights` keyword\\
 can also be a single (non-cell) array, in which case the weights are the same for all source positions:
     judiWeights(weights; nsrc=1)
 """
-function judiWeights(weights::Array; nsrc=1, vDT::DataType=Float32)
-    (eltype(weights) != vDT) && (weights = convert(Array{vDT},weights))
+function judiWeights(weights::Array{T, N}; nsrc=1, vDT::DataType=Float32) where {T, N}
+    (T != vDT) && (weights = convert(Array{vDT},weights))
     # length of vector
     n = 1
     m = prod(size(weights))*nsrc
-    weightsCell = Array{Array}(undef, nsrc)
-    for j=1:nsrc
-        weightsCell[j] = deepcopy(weights)
-    end
+    weightsCell = [deepcopy(weights) for j=1:nsrc]
     return judiWeights{vDT}("Extended source weights",m,n,nsrc,weightsCell)
 end
 
 # constructor if weights are passed as a cell array
-function judiWeights(weights::Union{Array{Any,1},Array{Array,1}}; vDT::DataType=Float32)
+function judiWeights(weights::Array{Array{T, N},1}; vDT::DataType=Float32) where {T, N}
```

Comment:
This change makes the extended source modeling example break here:
``` 
julia> for j=1:nsrc
           weights[j] = randn(Float32, model.n)
       end

julia> w = judiWeights(weights)
ERROR: MethodError: Cannot `convert` an object of type Array{Float32,2} to an object of type Float32
Closest candidates are:
  convert(::Type{Float32}, ::SegyIO.IBMFloat32) at /home/rorozcom3/.julia/packages/SegyIO/uw4ZW/src/types/IBMFloat32.jl:9
  convert(::Type{T}, ::T) where T<:Number at number.jl:6
  convert(::Type{T}, ::Number) where T<:Number at number.jl:7
  ...
Stacktrace:
 [1] setindex!(::Array{Float32,1}, ::Array{Float32,2}, ::Int64) at ./array.jl:847
 [2] _unsafe_copyto!(::Array{Float32,1}, ::Int64, ::Array{Array,1}, ::Int64, ::Int64) at ./array.jl:257
 [3] unsafe_copyto! at ./array.jl:311 [inlined]
 [4] _copyto_impl! at ./array.jl:335 [inlined]
 [5] copyto! at ./array.jl:321 [inlined]
 [6] copyto! at ./array.jl:347 [inlined]
 [7] copyto_axcheck! at ./abstractarray.jl:946 [inlined]
 [8] Array at ./array.jl:562 [inlined]
 [9] Array at ./boot.jl:431 [inlined]
 [10] convert(::Type{Array{Float32,N} where N}, ::Array{Array,1}) at ./array.jl:554
 [11] judiWeights(::Array{Array,1}; nsrc::Int64, vDT::DataType) at /home/rorozcom3/.julia/packages/JUDI/LqD3Y/src/TimeModeling/Types/judiWeights.jl:46
 [12] judiWeights(::Array{Array,1}) at /home/rorozcom3/.julia/packages/JUDI/LqD3Y/src/TimeModeling/Types/judiWeights.jl:46
 [13] top-level scope at REPL[35]:1

```
as per francis suggestion Looks like we need the addition of Array{Array{Any,1},1} in the multiple dispatch somewhere.

Reply:
No, Any is a bad type that's the whole point of that PR, types should be corrected where wrong and too generic.

Reply:
Basically the problem (found by @rafaelorozco) is on this line https://github.com/slimgroup/JUDI.jl/blob/dda83aa9971002241fe9e0fed6285f5f21cb7547/examples/scripts/modeling_extended_source_2D.jl#L58 in the example script

```julia
weights = Array{Array}(undef, nsrc)
for j=1:nsrc
    weights[j] = randn(Float32, model.n)
end
w = judiWeights(weights)
```

This gives error because
```julia
julia> typeof(weights)
Array{Array,1}
```

We either change the `judiWeights.jl` file as @rafaelorozco suggested above, or change the example script to use `weights = Array{Array{Float32,2}}(undef, nsrc)` as an initialization of the weights (I believe this enforces users to take more care of the weights).